### PR TITLE
ROC-2645: Move claim status to citizen frontend

### PR DESCRIPTION
**Pull request by jacmurphy**
### JIRA link
https://tools.hmcts.net/jira/browse/ROC-2645

### Change description
Light refactoring to include the use of claimStatus.
Moving the responseStatus.njk file to the citizen-frontend and renaming it as "status.njk"

### Developer self-QA run statement

- [x] I have clicked through the running application to see if all changes I made actually work.

### Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: src/main/features/dashboard/views/claimant.njk [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {% extends "layout.njk" %}
-{% from "responseStatus.njk" import responseStatusForClaimant %}
+{% from "main/features/dashboard/views/macros/status.njk" import statusForClaimant, claimantRespondToOffer %}
```
Comment (timja-kainos) [inline src/main/features/dashboard/views/claimant.njk]: I would have thought that `macros/status` would work as it should be in the search path

File: src/main/features/dashboard/views/claimant.njk [hunk 2]
```diff
@@ -16,7 +16,20 @@
...
+        </p>
+
+         {% macro claimantRespondToOffer (claim, url) %}
+          {% if claim.status === ClaimStatus.OFFER_SUBMITTED %}
+            {{ t('View <a href="{{ url }}">their offer</a>.',
+            { url : url  }) | safe }}
+          {% endif %}
+        {% endmacro %}
+
+
```
Comment (timja-kainos) [inline src/main/features/dashboard/views/claimant.njk]: remove extra white space here please

File: src/main/features/dashboard/views/claimant.njk [hunk 3]
```diff
@@ -16,7 +16,20 @@
     <div class="column-two-thirds">
       <div>
         <h2 class="heading-medium">{{ t('Claim status') }}</h2>
-        {{ responseStatusForClaimant(claim, DashboardUrlHelper.getStatusUrl(claim)) }}
+        {{ statusForClaimant(claim, DashboardUrlHelper.getStatusUrl(claim)) }}
+
+        <p>
```
Comment (timja-kainos) [inline src/main/features/dashboard/views/claimant.njk]: what is this?

Comment (jacmurphy) [reply src/main/features/dashboard/views/claimant.njk]: Its the same macro just renamed as it is more than just a status about a response

File: src/main/features/dashboard/views/claimant.njk (hunk 1 above)
Comment (jacmurphy) [reply src/main/features/dashboard/views/claimant.njk]: updating now

File: src/main/features/dashboard/views/defendant.njk [hunk 4]
```diff
@@ -16,7 +16,7 @@
 
       {{ responseStatusForDefendant(claim, DashboardUrlHelper.getStatusUrl(claim), isAfter4pm) }}
 
-      {% if featureToggles.offer and not claim.settlementReachedAt  %}
+      {% if featureToggles.offer and not ClaimStatus == OFFER_SUBMITTED and not ClaimStatus == OFFER_SETTLEMENT_REACHED  %}
```
Comment (timja-kainos) [inline src/main/features/dashboard/views/defendant.njk]: How does this work?

File: src/main/features/dashboard/views/index.njk [hunk 5]
```diff
@@ -1,6 +1,6 @@
 {% extends "layout.njk" %}
 {% from "form.njk" import submitButton, errorSummary %}
-{% from "responseStatus.njk" import responseStatusForDefendant, responseStatusForClaimant %}
+{% from "./macros/status.njk" import statusForDefendant, statusForClaimant %}
```
Comment (timja-kainos) [inline src/main/features/dashboard/views/index.njk]: 3 different ways of importing this macro :(

File: src/main/features/dashboard/views/claimant.njk (hunk 3 above)
Comment (timja-kainos) [reply src/main/features/dashboard/views/claimant.njk]: why is there a macro there twice then?

File: src/main/features/dashboard/views/defendant.njk (hunk 4 above)
Comment (jacmurphy) [reply src/main/features/dashboard/views/defendant.njk]: If you look into the claim.ts (src/main/app/claims/models/claim.ts) file, i already do that check. This change was for more consistency so we aren't using claim.settlementReachedAt() anymore and instead use OFFER_SETTLEMENT_REACHED

File: src/main/features/dashboard/views/index.njk (hunk 5 above)
Comment (jacmurphy) [reply src/main/features/dashboard/views/index.njk]: i was having issues with it and got one that worked and just passed my mind. Updating now

File: src/main/features/dashboard/views/claimant.njk (hunk 2 above)
Comment (jacmurphy) [reply src/main/features/dashboard/views/claimant.njk]: 👍

File: src/main/features/dashboard/views/defendant.njk (hunk 4 above)
Comment (timja-kainos) [reply src/main/features/dashboard/views/defendant.njk]: but isn't `ClaimStatus` just a global variable? this check doesn't seem to do anything

File: src/main/features/dashboard/views/claimant.njk (hunk 3 above)
Comment (jacmurphy) [reply src/main/features/dashboard/views/claimant.njk]: Are you referring to "claimantRespondToOffer"?
I don't see it above in your highlight.

File: src/main/features/dashboard/views/index.njk (hunk 5 above)
Comment (jacmurphy) [reply src/main/features/dashboard/views/index.njk]: changed

File: src/main/features/dashboard/views/claimant.njk (hunk 3 above)
Comment (timja-kainos) [reply src/main/features/dashboard/views/claimant.njk]: yes

File: src/main/features/dashboard/views/defendant.njk (hunk 4 above)
Comment (jacmurphy) [reply src/main/features/dashboard/views/defendant.njk]: It was my understanding that it checks if the toggle offer is on, it makes sure OFFER_SUBMITTED *for that offer* is not true and OFFER_SETTLEMENT_REACHED *for that offer* is not true and then doesn't give them the option to create an offer.

If you have an offer that has been accepted and remove that code, the defendant can still make an offer

Comment (timja-kainos) [reply src/main/features/dashboard/views/defendant.njk]: how does this code work though? what is the `ClaimStatus` variable?

File: src/main/features/dashboard/views/claimant.njk (hunk 3 above)
Comment (jacmurphy) [reply src/main/features/dashboard/views/claimant.njk]: I understand now.
statusForClaimant is the status that a claimant will see in the dashboard and the claimant.njk.
claimantRespondToOffer is the clickable "View their offer." link in the claimant page (That was removed from the status)

File: src/main/features/dashboard/views/defendant.njk (hunk 4 above)
Comment (jacmurphy) [reply src/main/features/dashboard/views/defendant.njk]: If you look into the claim.ts (src/main/app/claims/models/claim.ts) i validate the conditions that the claim falls under.
For example 
```
 } else if (this.eligibleForCCJ) {
      return ClaimStatus.ELIGIBLE_FOR_CCJ
```

```
get eligibleForCCJ (): boolean {
    if (!toBoolean(config.get<boolean>('featureToggles.countyCourtJudgment'))) {
      return false
    }
```

**Pull request closed without merging**